### PR TITLE
compiler-cli: add prelink option and setup dual compliance tests

### PR DESCRIPTION
**Pull request by petebacondarwin**
Another alternative for dual compliance testing

### CODE REVIEW

File: packages/compiler-cli/test/compliance/r3_compiler_compliance_spec.ts [hunk 1]
```diff
@@ -8,14 +8,22 @@
...
  * test in compiler_canonical_spec.ts should have a corresponding test here.
  */
 describe('compiler compliance', () => {
+  const context = setupContext();
+
+  let compile: ReturnType<typeof createCompileFn>;
+  beforeAll(() => {
+    compile = createCompileFn(context.compilationMode);
+  });
+
```
Comment (petebacondarwin) [inline packages/compiler-cli/test/compliance/r3_compiler_compliance_spec.ts:26]: This block would need to be added to each compliance spec file.

File: packages/compiler-cli/test/compliance/mock_compile/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,19 @@
+load("//tools:defaults.bzl", "ts_library")
+
+ts_library(
+    name = "mock_compile",
+    testonly = 1,
```
Comment (JoostK) [inline packages/compiler-cli/test/compliance/mock_compile/BUILD.bazel:5]: I'm seeing both `True` and `1` being used, but `True` makes more sense to me:

```suggestion
    testonly = True,
```

File: packages/compiler-cli/test/compliance/prelink/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,29 @@
+load("//tools:defaults.bzl", "jasmine_node_test", "ts_library")
+
+ts_library(
+    name = "prelink_bootstrap",
+    testonly = 1,
```
Comment (JoostK) [inline packages/compiler-cli/test/compliance/prelink/BUILD.bazel:5]:
```suggestion
    testonly = True,
```

**Review by JoostK — APPROVED** [review]

**Comment by JoostK** [comment]
@googlebot I consent.

File: packages/compiler-cli/test/compliance/mock_compile/BUILD.bazel (hunk 2 above)
Comment (petebacondarwin) [reply packages/compiler-cli/test/compliance/mock_compile/BUILD.bazel:5]: Indeed that seems correct https://docs.bazel.build/versions/master/be/common-definitions.html

**Review by alxhub — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:334188134:BASE:334188163:1601314494402:b474077d).

**Comment by AndrewKushnir** [comment]
FYI, presubmit is successful for the changes in this PR.

**Comment by petebacondarwin** [comment]
Thanks @AndrewKushnir !

**Pull request closed without merging**